src/porepy/viz/exporter.py

```diff
@@ -180,6 +182,183 @@ def __init__(
+            to the grid size. If possible, translate the value to a scalar-ranged
+            object.
+
+            Args:
+                value (np.ndarray): input array to be converted
+                grid (pp.Grid or pp.MortarGrid): subdomain or interface to which value
+                    is associated to
+
+            Raises:
+                ValueError if the value array is not compatible with the grid
+            """
+            # Make some checks
+            if not value.size % grid.num_cells == 0:
+                # This line will raise an error if node or face data is exported.
+                raise ValueError("The data array is not compatible with the grid.")
+
+            # Apply flatten with the order compatible with _toVectorFormat.
+            return value.flatten("F")
+
+        # Convert file_names to a list
+        if isinstance(file_names, str):
+            file_names = [file_names]
+
+        # Consider each file separately.
+        # Procedure has three steps:
+        # 1. Determine dimensionality and type of grid, addressed by the vtu file.
+        # 2. Check whether the vtu file is compatible with the corresponding grids.
+        # 3. Transfer data from vtu to the mixed-dimensional grid.
+        for i, file_name in enumerate(file_names):
+
+            # Read all data from vtu
+            vtu_data = meshio.read(file_name)
+
+            # 1. Step: Determine dimension of the grid associated to vtu file, and
+            # whether the grid corresponds to a subdomain or interface.
+
+            # Allow for automatic detection from the file name.
+            if kwargs.pop("automatic", True):
+                # Assuming grid_dim and is_mortar are among the stored vtu data
+                # one could utilize those, but this does not have to be the case.
+                # Thus, use naming convention of the Exporter for this.
+
+                # Remove ending '.vtu' from file_name, and decompose into pieces
+                # separated by '_'.
+                file_name_pieces = Path(file_name).stem.split("_")
+
+                # If the last two are numbers, the first one denotes the dimension.
+                assert file_name_pieces[-1].isnumeric()
+                dim_pos = -2 if file_name_pieces[-2].isnumeric() else -1
+                dim = int(file_name_pieces[dim_pos])
+
+                # If the key words before are 'mortar', or 'mortar' and 'constant',
+                # the vtu file corresponds to intefaces; otherwise to subdomains.
+                is_subdomain_data = not (
+                    file_name_pieces[dim_pos - 1] == "mortar"
+                    or file_name_pieces[dim_pos - 1] == "constant"
+                    and file_name_pieces[dim_pos - 2] == "mortar"
+                )
+
+            else:
+                # Read from keyword arguments
+
+                # Dimensionality of the grid
+                if "dims" not in kwargs:
```

Comment:
I think this effectively means dims should not be a kwargs, but a standard argument.

Reply:
`dims` is only required when all required properties like dimensionality of the grid and nature (mortar or subdomain) corresponding to the vtu file are not deducted automatically. See my comment above on the automatic deduction.

Do you still think, I should promote `dims` to be a standard argument and set it to None by default?

BTW: I would be happy to hear your opinion on how to distribute arguments to standard, **args and **kwargs arguments!

Reply:
I understand, I'm fine with this solution.